src/main/java/spireMapOverhaul/zones/heavenlyClouds/HeavenlyCloudsZone.java

```diff
@@ -0,0 +1,60 @@
+package spireMapOverhaul.zones.heavenlyClouds;
+
+import basemod.helpers.CardModifierManager;
+import com.badlogic.gdx.graphics.Color;
+import com.megacrit.cardcrawl.actions.common.ApplyPowerAction;
+import com.megacrit.cardcrawl.cards.AbstractCard;
+import com.megacrit.cardcrawl.dungeons.AbstractDungeon;
+import com.megacrit.cardcrawl.monsters.AbstractMonster;
+import spireMapOverhaul.abstracts.AbstractZone;
+import spireMapOverhaul.util.Wiz;
+import spireMapOverhaul.zoneInterfaces.CombatModifyingZone;
+import spireMapOverhaul.zoneInterfaces.RewardModifyingZone;
+
+import java.util.ArrayList;
+
+import static spireMapOverhaul.util.Wiz.atb;
+
+public class HeavenlyCloudsZone extends AbstractZone implements CombatModifyingZone, RewardModifyingZone {
```

Comment:
Two edge cases are worth thinking about:
1. What happens to the Byrds fight? It would be pretty weird to see them have flight twice. Given how flight is their thing, special casing how this zone interacts with Byrds makes the most sense to me.
2. What about newly spawned enemies (e.g. Gremlin Leader's gremlins)? We don't have a hook for this yet, and it's fine to leave it out, but I wanted to make sure you were happy with how this zone worked with that.

Reply:
Ignoring minions is a sensible way to do it. One thought that occurred to me is that I'd want the logic to ignore minions that are present at the start of combat too (e.g. for the Gremlin Leader fight). It would be a bit strange if the initially present minions got Flight, but newly spawned ones didn't. With that approach something like Gremlin Leader would count as a single-enemy fight too (since only one enemy is getting flight). That's what makes the most sense to me, at least.

Reply:
Byrds in the flight zone now get +1 strength and a line of trash talk. Minions are excluded from the flight mechanic.

The only other problematic enemy I noticed with my zone is Spheric Guardian and to a lesser extent Sentry due to their artifact allowing them to block the Stun from Flight's removal and Spheric Guardian's extremely low hp pool effectively negating the hp reduction of Flight. I don't really know of an elegant solution to this issue beyond hardcoding an exception for Spheric Guardian in particular.

Reply:
Maybe have it also reduce the block enemies start with? Hopefully the timing works out so that you can catch Spheric Guardian's block that way. That isn't perfect since a lot of Spheric Guardian's "HP" is the block it gains during the fight, but it seems close enough.

For the Stun thing, it's weird to me that Stun is a debuff at all. When it's used for a mechanic like this, it shouldn't be. I'd support making your own version of `StunMonsterAction` that does something like:
```
StunMonsterPower power = new new StunMonsterPower(this.target, this.amount);
power.type = NeutralPowertypePatch.NEUTRAL;
AbstractDungeon.actionManager.addToTop(new ApplyPowerAction(this.target, this.source, power, this.amount));
```

Reply:
Updated to nerf spheric guardian's start of combat block and make stun go past artifact